src/main/java/com/google/sps/servlets/FindMentorServlet.java

```diff
@@ -109,19 +109,26 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     boolean success = false;
 
-    User user = dataAccess.getCurrentUser();
-    if (user != null) {
-      Mentee mentee = dataAccess.getMentee(user.getUserId());
-      Mentor mentor = dataAccess.getMentor(Long.parseLong(mentorKey));
-      if (mentee != null && mentor != null) {
-        if (choice.equals(SEND)) {
-          MentorshipRequest mentorshipRequest =
-              new MentorshipRequest(mentor.getDatastoreKey(), mentee.getDatastoreKey());
-          dataAccess.publishRequest(mentorshipRequest);
-          success = true;
-        } else if (choice.equals(DISLIKE)) {
-          dataAccess.dislikeMentor(mentee, mentor);
-          success = true;
+    Long mentorDatastoreKey = null;
+    try {
+      mentorDatastoreKey = Long.parseLong(mentorKey);
+    } catch (NumberFormatException e) {
+    }
+    if (mentorDatastoreKey != null && (choice.equals(SEND) || choice.equals(DISLIKE))) {
+      User user = dataAccess.getCurrentUser();
+      if (user != null) {
+        Mentee mentee = dataAccess.getMentee(user.getUserId());
+        Mentor mentor = dataAccess.getMentor(mentorDatastoreKey);
+        if (mentee != null && mentor != null) {
```

Comment:
same here, as a rule of thumb after you have more than 2 if nests consider if you can re-work the statements